pkg/login/social/okta_oauth.go

```diff
@@ -74,10 +74,13 @@ func (s *SocialOkta) UserInfo(client *http.Client, token *oauth2.Token) (*BasicU
 	var data OktaUserInfoJson
 	err = s.extractAPI(&data, client)
 	if err != nil {
-		s.log.Error("Error extracting user info", "error", err)
+		return nil, err
 	}
 
-	role := s.extractRole(&data)
+	role, err := s.extractRole(&data)
+	if err != nil {
+		s.log.Error("Failed to extract role", "error", err)
```

Comment:
```suggestion
		return nil, err
```

Reply:
@marefr What do you think about this?
>Arve Knudsen 
The question is if we should be swallowing unexpected errors, that doesn't sound quite right to me
"no role attribute found" for example is an expected error, and that's fine. We shouldn't even log it as an error (edited) 
However, what about the case where there's for example a progamming or an I/O error. What's the right approach then? Just to log the error? Or should we bubble the error up?

>Alex
extractRole() is using for role mapping (map custom attribute from auth server to Grafana role, like viewer, editor or admin). Default role is "Viewer". So if no role attribute found or any error happened, user will be logged in with en empty role (Viewer)

Reply:
For context, I think the main question is if we should stop unexpected errors (e.g. programming errors, I/O errors) from bubbling up and just log them here. I know too little about the OAuth flow to tell.

Reply:
>if we should stop unexpected errors (e.g. programming errors, I/O errors) from bubbling up and just log them here

I think yes, because otherwise user won't log in in cases user could before.

Reply:
@alexanderzobnin is correct. We have to use logging in certain places instead of bubbling up errors.

Since this part of the code base is a bit messy and easy to introduce regressions I would suggest handling further (bigger) refactorings/suggestions separately from this PR.

Reply:
Yes, I feel the same. In this PR I tried to do not touch old code too much because of potential regression.